metadata-ingestion/src/datahub/ingestion/source/tableau.py

```diff
@@ -1948,19 +2088,25 @@ def emit_table(
         else:
             logger.debug(f"Browse path not set for table {database_table.urn}")
 
-        schema_metadata = self.get_schema_metadata_for_table(columns or [])
+        schema_metadata = self.get_schema_metadata_for_table(
+            tableau_columns, database_table.parsed_columns
+        )
         if schema_metadata is not None:
             dataset_snapshot.aspects.append(schema_metadata)
 
         yield self.get_metadata_change_event(dataset_snapshot)
 
     def get_schema_metadata_for_table(
-        self, columns: List[dict]
+        self,
+        tableau_columns: Optional[List[Dict[str, Any]]],
+        parsed_columns: Optional[Set[str]] = None,
     ) -> Optional[SchemaMetadata]:
         schema_metadata: Optional[SchemaMetadata] = None
-        if columns:
-            fields = []
-            for field in columns:
+
+        fields = []
```

Comment:
if you make fields a dict of column name -> SchemaField, the code below can be more readable e.g.

```
for parsed_column in parsed_columns or []:
	column_name = ...
	if column_name in fields:
		continue
	fields.append(SchemaField(...))
```

Reply:
it's more a matter of taste. could we keep it as is?